src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -66,6 +66,22 @@ function CartesianPoint(x::TX, y::TY, z::TZ) where {TX<:Real,TY<:Real,TZ<:Real}
     CartesianPoint{T}(T(x),T(y),T(z))
 end
 
+#Unit support
+function CartesianPoint(x::Unitful.Quantity, y::Unitful.Quantity, z::Unitful.Quantity)
+    for (i, pt) in enumerate((x, y, z))
+        !(pt isa Unitful.Length) && throw(ArgumentError(
+            "Coordinate $i must be a length, got $(pt) with unit $(Unitful.unit(pt))"
+        ))
+    end
+    CartesianPoint(x::Unitful.Length, y::Unitful.Length, z::Unitful.Length)
+end
+
+function CartesianPoint(x::Unitful.Length, y::Unitful.Length, z::Unitful.Length)
+    vals = to_internal_units.( (x, y, z) )
+    T = all(v -> v isa Float32, vals) ? Float32 : Float64
+    CartesianPoint{T}(T.(vals)...)
+end
+
```

Comment:
This right now doesn't cover the case `CartesianPoint(1u"mm", 0, 0)` (if only some of the arguments have units.
Consider making this equivalent to what we did for `CylindricalPoint`:
```suggestion
function CartesianPoint(x, y, z)
    for (name, pt) in zip((:x,:y,:z), (x, y, z))
        (pt isa Real || pt isa Unitful.Length) || throw(ArgumentError(
            "Expected `$(name)` to be a length or Real, got unit $(Unitful.unit(pt))"
        ))
    end
    x_val = to_internal_units(x)
    y_val = to_internal_units(y)
    z_val = to_internal_units(z)
    return CartesianPoint(x_val, y_val, z_val)
end
```